### PR TITLE
Gradle and Checkstyle updates

### PR DESCRIPTION
**Pull request by rozza**
In my POJO branch I was hitting: https://github.com/checkstyle/checkstyle/issues/2141. So I looked at upgrading checkstyle and gradle.  This brings us up to date for gradlew on the 2.x branch and the latest checkstyle.

The first commit is the upgrade and the second is all the fix for all the new checkstyle issues.

### CODE REVIEW

**Comment by evanchooly** [comment]
not to gradle 3.3?

lgtm

**Comment by rozza** [comment]
Gradle 3.3 would be nice - but theres some hoops for testing Java 6 (see https://docs.gradle.org/current/userguide/java_plugin.html#sec:java_cross_compilation)

I'm not sure how well that will play with evergreen - so I didn't want to bump it for this.

File: driver/build.gradle [hunk 1]
```diff
@@ -43,7 +43,11 @@ clirr {
 
 tasks.withType(Checkstyle) {
     // needed so the Javadoc checks can find the code in other modules
-    classpath = files(project(':driver-core').sourceSets.main.output, sourceSets.main.output);
+    classpath = files(project(':driver-core').sourceSets.main.output, sourceSets.main.output)
+    reports {
```
Comment (jyemin) [inline driver/build.gradle]: Why is this reports section necessary, and if it is, why don't we need it for bson and driver-core?

Comment (rozza) [inline driver/build.gradle]: it makes it much easier to read the checkstyle errors.  Weirdly it works for the bson section even though its not defined there - I'll move to the common area.

Comment (rozza) [inline driver/build.gradle]: Done.

**Comment by jyemin** [comment]
LGTM

**Pull request closed without merging**